File: web/src/main/java/uk/ac/ebi/phenodigm/model/Allele.java
/*
 * Copyright © 2011-2013 EMBL - European Bioinformatics Institute
 * and Genome Research Limited
 *  
 * Licensed under the Apache License, Version 2.0 (the "License"); 
 * you may not use this file except in compliance with the License.  
 * You may obtain a copy of the License at
 * 
 *      http://www.apache.org/licenses/LICENSE-2.0
 * 
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package uk.ac.ebi.phenodigm.model;

import java.util.Objects;

/**
 * Class for holding allele data.
 * @author Jules Jacobsen <jules.jacobsen@sanger.ac.uk>
 * 
 * @deprecated pdsimplify
 */
public class Allele {

    private final String source;
    private final String geneId;
    private final String alleleSymbol;
    private final String alleleId;

    public Allele(String source, String geneId, String alleleSymbol, String alleleId) {
        this.source = source;
        this.geneId = geneId;
        this.alleleSymbol = alleleSymbol;
        this.alleleId = alleleId;
    }

    public String getSource() {
        return source;
    }

    public String getGeneId() {
        return geneId;
    }

    public String getAlleleSymbol() {
        return alleleSymbol;
    }

    public String getAlleleId() {
        return alleleId;
    }

    @Override
    public int hashCode() {
        int hash = 7;
        hash = 67 * hash + Objects.hashCode(this.source);
        hash = 67 * hash + Objects.hashCode(this.geneId);
        hash = 67 * hash + Objects.hashCode(this.alleleSymbol);
        hash = 67 * hash + Objects.hashCode(this.alleleId);
        return hash;
    }

    @Override
    public boolean equals(Object obj) {
        if (obj == null) {
            return false;
        }
        if (getClass() != obj.getClass()) {
            return false;
        }
        final Allele other = (Allele) obj;
        if (!Objects.equals(this.source, other.source)) {
            return false;
        }
        if (!Objects.equals(this.geneId, other.geneId)) {
            return false;
        }
        if (!Objects.equals(this.alleleSymbol, other.alleleSymbol)) {
            return false;
        }
        return Objects.equals(this.alleleId, other.alleleId);
    }

    @Override
    public String toString() {
        return "Allele{" + "source=" + source + ", geneId=" + geneId + ", alleleId=" + alleleId + ", alleleSymbol=" + alleleSymbol + '}';
    }
}

File: web/src/main/webapp/resources/js/general/allele.js
/**
 * Copyright © 2011-2013 EMBL - European Bioinformatics Institute
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * toggle: used for the image dropdown.
 *
 */
jQuery(document).ready(	function() {

        function getURLParameter(name) {
          return decodeURIComponent((new RegExp('[?|&]' + name + '=' + '([^&;]+?)(&|#|;|$)').exec(location.search)||[,""])[1].replace(/\+/g, '%20'))||null
        }

//        try {
//            $.ajax({
//                    url: baseUrl + '/genesAllele2/' + gene_id,
//
//                    timeout: 5000,
//                    success: function (response) {
//
//                        $('#allele2').html(response);
//                    }
//                    ,error: function(x, t, m) {
//                        var errorMsg='<td class="gene-data" id="allele_links"><font color="red"><font color="red">Error trying to retrieve allele product infomation</font></td>';
//                        $('#allele2').html(errorMsg);
//                    }
//            });
//        }
//        catch(err) {
//            console.log("#### Exception: " + err);
//        }
        
//        try {
//            $.ajax({
//                    url: baseUrl + '/orderSection?acc=' + gene_id,
//
//                    timeout: 5000,
//                    success: function (response) {
//
//                        $('#order_product').html(response);
//                    }
//                    ,error: function(x, t, m) {
//                        var errorMsg='<td class="gene-data" id="allele_links"><font color="red"><font color="red">Error trying to retrieve allele product infomation</font></td>';
//                        $('#order_product').html(errorMsg);
//                    }
//            });
//        }
//        catch(err) {
//            console.log("#### Exception: " + err);
//        }

        $('.qcData').each(function(){
            var type = $(this).data("type");
            var name = $(this).data("name");
            var alleleType = $(this).data("alleletype");

            var bare_str = "";
            if (bare === "true"){ bare_str = "&bare=" + bare;}
            var url = baseUrl + '/alleles/qc_data/' + type + '/' + name + "?simple=true" + bare_str; 

            if( ! type || ! name) {

                $(this).html('<p>Not found!</p>');
                return;
            }

        	$.ajax({
		    url: url,
		    timeout: 2000,
                    context: this,
		    success: function (response) {

                        try {
                            $(this).html(response);
                        }
                        catch(err) {
                            console.log("#### Exception: " + err);
                        }

		    }
		    ,error: function(x, t, m) {
			var errorMsg='<td>QC Data Link:</td><td class="gene-data" id="allele_links"><font color="red"><font color="red">Error trying to retrieve QC Data infomation</font></td>';
	    	        $(this).html(errorMsg);
	            }
	        });
        });

        $(".hasTooltip").each(function(){
            $(this).qtip({
                content: {text: $(this).next('div')},
                position: {
			my: 'top middle',
			at: 'bottom middle'
			},
                style: { classes: 'ui-tooltip-wideimage'
                     }
            });
        });

    function toggleTable(id) {
        var target = "#" + id + "_toggle";
        $(target).on({'click':function(event){
        event.preventDefault();
        $("#" + id + " .rest").toggle("fast");

        if($(target).hasClass("toggle_closed")) {
            $(target).removeClass("toggle_closed");
            $(target).addClass("toggle_open");
            var type = $(target).data( "type" );
            $(target).text("Hide " + type);
        }
        else {
            $(target).removeClass("toggle_open");
            $(target).addClass("toggle_closed");
            var type = $(target).data( "type" );
            var count = $(target).data( "count" );
            $(target).text("Show all " + count + " " + type);
        }
        }});
    }

    toggleTable("mouse_table");
    toggleTable("es_cell_table");
    toggleTable("targeting_vector_table");

    $("#mice_order_contact_button").on({'click':function(event){
        if($("#mouse_table_toggle").hasClass("toggle_closed")) {
            $( "#mouse_table_toggle" ).trigger( "click" );
        }
    }});

    $("#es_cell_order_contact_button").on({'click':function(event){
        if($("#es_cell_table_toggle").hasClass("toggle_closed")) {
            $( "#es_cell_table_toggle" ).trigger( "click" );
        }
    }});

});